File: win-prog/ProgWinCS/Text and Fonts/NaiveFamiliesList/NaiveFamiliesList.cs
//------------------------------------------------
// NaiveFamiliesList.cs © 2001 by Charles Petzold
//------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class NaiveFamiliesList: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new NaiveFamiliesList());
     }
     public NaiveFamiliesList()
     {
          Text = "Naive Font Families List";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Brush        brush = new SolidBrush(clr);
          float        y     = 0;
          FontFamily[] aff   = FontFamily.Families;

          foreach (FontFamily ff in aff)
          {
               Font font = new Font(ff, 12);
               grfx.DrawString(ff.Name, font, brush, 0, y);
               y += font.GetHeight(grfx);
          }
     }
}

File: win-prog/ProgWinVB/Text and Fonts/NaiveFamiliesList/NaiveFamiliesList.vb
'--------------------------------------------------
' NaiveFamiliesList.vb (c) 2002 by Charles Petzold
'--------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class NaiveFamiliesList
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New NaiveFamiliesList())
    End Sub

    Sub New()
        Text = "Naive Font Families List"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim br As New SolidBrush(clr)
        Dim y As Single = 0
        Dim aff() As FontFamily = FontFamily.Families
        Dim ff As FontFamily

        For Each ff In aff
            'Try
            Dim fnt As New Font(ff, 12)
            grfx.DrawString(ff.Name, fnt, br, 0, y)
            y += fnt.GetHeight(grfx)
            'Catch
            'End Try
        Next ff
    End Sub
End Class
